From dd4e712ba7605e97e2f55fb997085951c1fcd7d2 Mon Sep 17 00:00:00 2001
From: Robby Cai <R63905@freescale.com>
Date: Mon, 18 Apr 2011 15:21:46 +0800
Subject: [PATCH] ENGR00142259 set UART2_RXD (GP6_11) to high

Isolate EIM signals and boot configuration signals.
Without this setting, the chip's temperature will be high.

Signed-off-by: Robby Cai <R63905@freescale.com>
---
 board/freescale/mx50_rdp/mx50_rdp.c |   10 ++++++++++
 1 files changed, 10 insertions(+), 0 deletions(-)

diff --git a/board/freescale/mx50_rdp/mx50_rdp.c b/board/freescale/mx50_rdp/mx50_rdp.c
index a00f6c9..5bac73a 100644
--- a/board/freescale/mx50_rdp/mx50_rdp.c
+++ b/board/freescale/mx50_rdp/mx50_rdp.c
@@ -663,6 +663,16 @@ static void setup_fec(void)
 	reg = readl(GPIO4_BASE_ADDR + 0x4);
 	reg |= (1 << 16);
 	writel(reg, GPIO4_BASE_ADDR + 0x4);
+
+	/* Isolate EIM signals and boot configuration signals. - GPIO6_11 to 1*/
+	mxc_request_iomux(MX50_PIN_UART2_RXD, IOMUX_CONFIG_ALT1);
+	reg = readl(GPIO6_BASE_ADDR + 0x0);
+	reg |= (1 << 11);
+	writel(reg, GPIO6_BASE_ADDR + 0x0);
+
+	reg = readl(GPIO6_BASE_ADDR + 0x4);
+	reg |= (1 << 11);
+	writel(reg, GPIO6_BASE_ADDR + 0x4);
 #endif
 
 	/*FEC_MDIO*/
-- 
1.5.4.4

